websocket.go

```diff
@@ -143,19 +185,29 @@ func (c *Connection) handleResponse() {
 		c.clientMap.lock.RLock()
 		c.subscriberMap.lock.RLock()
 		if c.clientMap.clients[r.Id] != nil {
+			c.closeWg.Add(1)
```

Comment:
Just a note: waitgroups don't seem to be designed to be used for short-lived processes, as, if the waitgroup counter increments while waiting, it can cause panics (see https://play.golang.org/p/y-6RylTR69L). Just envisioning problems if e.g. `c.closeWg` is waited on in the middle of this executing.

Reply:
Yep, makes sense - I wrote a test case for this in go playground and seemed OK. Just select on the closeSig and should be fine?

Reply:
Just concerned about threadsafety, depends when `c.Close` is called? Could be between when we select on `closeSig` and when we add to the waitgroup. Or is there some other order-enforcement that I'm missing?

Reply:
Nope, I think you're right. I also think there is not really any risk, as those channel reads and writes are guarded by reading from `c.closeSig` anyway, so should all just give up and exit very quickly.

Reply:
Have pushed a change removing the waitgroup logic from the message writing logic.